build.rs

```diff
@@ -153,8 +153,9 @@ fn main() {
     ];
     write_enum_variants_to_file(&mut file, "COMPRESSION_TYPES", libpng_compression_types);
 
-    write_constant_to_file(&mut file, "DEFAULT_THREAD_POOL_SIZE", num_cpus::get());
-    write_constant_to_file(&mut file, "MAX_THREAD_POOL_SIZE", num_cpus::get() * 10);
+    let threads = std::thread::available_parallelism().unwrap().get();
+    write_constant_to_file(&mut file, "DEFAULT_THREAD_POOL_SIZE", threads);
+    write_constant_to_file(&mut file, "MAX_THREAD_POOL_SIZE", threads * 10);
```

Comment:
Is performance improved when using the `MAX_THREAD_POOL_SIZE = threads * 10` compared to using `MAX_THREAD_POOL_SIZE = threads` ?

Reply:
Should be better. Having X cores means we can do X work in parallel but if we can have a couple more threads to swap while some other is stopped its usually good. Not necessarily 10x but this is just the maximum number we allow, not the actual number we set, so I think it is fine.